extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.settings.js

```diff
@@ -0,0 +1,193 @@
+
+		this.player.addButton(settingsIcon, 'Settings', function () {
+			if (!self.wikiaSettingsElement.style.display) {
+				self.open();
+			} else {
+				self.close();
+			}
+		}, this.buttonID, 'wikia-jw-settings-button');
+	};
+
+	WikiaJWPlayerSettings.prototype.removeButton = function () {
+		this.player.removeButton(this.buttonID);
+	};
+
+	WikiaJWPlayerSettings.prototype.close = function () {
+		this.showSettingsList();
+		this.wikiaSettingsElement.style.display = null;
+		this.player.getContainer().classList.remove('wikia-jw-settings-open');
+	};
+
+	WikiaJWPlayerSettings.prototype.open = function () {
+		this.wikiaSettingsElement.style.display = 'block';
+		this.player.getContainer().classList.add('wikia-jw-settings-open');
+	};
+
+	WikiaJWPlayerSettings.prototype.hide = function () {
+		this.close();
+		this.removeButton();
+	};
+
+	WikiaJWPlayerSettings.prototype.show = function () {
+		this.addButton();
+	};
+
+	WikiaJWPlayerSettings.prototype.showQualityLevelsList = function () {
+		this.settingsList.style.display = 'none';
+		this.qualityLevelsList.style.display = 'block';
+	};
+
+	WikiaJWPlayerSettings.prototype.showSettingsList = function () {
+		this.settingsList.style.display = 'block';
+		this.qualityLevelsList.style.display = 'none';
+	};
+
+	WikiaJWPlayerSettings.prototype.addSettingsContent = function (div) {
+		div.classList.add('wikia-jw-settings');
+		div.classList.remove('jw-reset', 'jw-plugin');
+		this.settingsList = this.createSettingsListElement();
+		this.qualityLevelsList = this.createQualityLevelsList();
+
+		div.appendChild(this.settingsList);
+		div.appendChild(this.qualityLevelsList);
+
+		return div;
+	};
+
+	WikiaJWPlayerSettings.prototype.createSettingsListElement = function () {
+		var settingsList = document.createElement('ul');
+
+		settingsList.classList.add('wikia-jw-settings__list');
+		settingsList.appendChild(this.createQualityButton());
+		settingsList.appendChild(this.createAutoplayToggle());
+
+		return settingsList;
+	};
+
+	WikiaJWPlayerSettings.prototype.createQualityButton = function () {
+		var qualityButton = document.createElement('li'),
+			self = this;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the declaration of the unused 'self' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "self" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)